modules/dnn/src/dnn.cpp

```diff
@@ -729,14 +729,41 @@ struct DataLayer : public Layer
         }
     }
 
-    virtual Ptr<BackendNode> initInfEngine(const std::vector<Ptr<BackendWrapper> >&) CV_OVERRIDE
+    virtual Ptr<BackendNode> initInfEngine(const std::vector<Ptr<BackendWrapper> >&,
+                                           const std::vector<Ptr<BackendNode> >& nodes) CV_OVERRIDE
     {
 #ifdef HAVE_INF_ENGINE
         CV_CheckEQ(inputsData.size(), (size_t)1, "");
         CV_CheckEQ(inputsData[0].dims, 4, "");
         const size_t numChannels = inputsData[0].size[1];
         CV_Assert(numChannels <= 4);
 
+#ifdef HAVE_DNN_NGRAPH
+        if (DNN_BACKEND_INFERENCE_ENGINE_NGRAPH == getInferenceEngineBackendTypeParam())
```

Comment:
1. Using `getInferenceEngineBackendTypeParam()` is not correct - it might be changed on the fly. `== preferableBackend` should be used instead, but it is not exposed for Layers.
2. `#ifdef HAVE_DNN_NGRAPH` should be used.